insights/tests/datasources/test_candlepin_broker.py

```diff
@@ -0,0 +1,170 @@
+    </core>
+</configuration>
+""".strip()
+
+CANDLEPIN_BROKER_BAD = """
+<config>
+    <core/>
+</configuration>
+""".strip()
+
+
+CANDLEPIN_BROKER_XML = """
+<configuration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:activemq /schema/artemis-configuration.xsd">
+    <core xsi:schemaLocation="urn:activemq:core ">
+        <security-enabled>true</security-enabled>
+        <security-settings>
+            <security-setting match="katello.candlepin.#">
+                <permission roles="candlepinEventsConsumer" type="consume" />
+            </security-setting>
+            <security-setting match="#">
+                <permission roles="invm-role" type="createAddress" />
+                <permission roles="invm-role" type="deleteAddress" />
+                <permission roles="invm-role" type="createDurableQueue" />
+                <permission roles="invm-role" type="deleteDurableQueue" />
+                <permission roles="invm-role" type="createNonDurableQueue" />
+                <permission roles="invm-role" type="deleteNonDurableQueue" />
+                <permission roles="invm-role" type="send" />
+                <permission roles="invm-role" type="consume" />
+                <permission roles="invm-role" type="browse" />
+                <permission roles="invm-role" type="manage" />
+            </security-setting>
+        </security-settings>
+        <cluster-user />
+        </core>
+</configuration>
+""".strip()
+
+
+CANDLE_BROKER_NO_SENTISVE_INFO = """
+<configuration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="urn:activemq /schema/artemis-configuration.xsd">
+    <core xsi:schemaLocation="urn:activemq:core ">
+        <security-enabled>true</security-enabled>
+        <security-settings>
+            <security-setting match="katello.candlepin.#">
+                <permission roles="candlepinEventsConsumer" type="consume" />
+            </security-setting>
+            <security-setting match="#">
+                <permission roles="invm-role" type="createAddress" />
+                <permission roles="invm-role" type="deleteAddress" />
+                <permission roles="invm-role" type="createDurableQueue" />
+                <permission roles="invm-role" type="deleteDurableQueue" />
+                <permission roles="invm-role" type="createNonDurableQueue" />
+                <permission roles="invm-role" type="deleteNonDurableQueue" />
+                <permission roles="invm-role" type="send" />
+                <permission roles="invm-role" type="consume" />
+                <permission roles="invm-role" type="browse" />
+                <permission roles="invm-role" type="manage" />
+            </security-setting>
+        </security-settings>
+        <cluster-user />
+    </core>
+</configuration>
+""".strip()
+
+RELATIVE_PATH = '/etc/candlepin/broker.xml'
+
+
+def test_candlepin_broker():
+    simple_file.content = CANDLEPIN_BROKER.splitlines()
```

Comment:
@huali027 this needs to be changed.  Its also wrong in cloud_init.  This may actually cause some side effects to the base `simple_file` object.  You can either create a new simple_file object and then set the content, or just create a new class that only has a content attribute.  You could also use a mock class.